historical_system_profiles/db_interface.py

```diff
@@ -58,6 +64,11 @@ def get_hsps_by_profile_ids(profile_ids, account_number):
     )
 
     query_results = query.all()
+    try:
+        ensure_correct_system_count(profile_ids, query_results)
```

Comment:
we should break `ensure_correct_system_count` out from the inventory service interface and move it elsewhere, perhaps view_helpers in kerlescan. However, that's outside the scope of this PR